File: evidence/generated-systems/golang/fintech-on-golang/fintech-on-golang/internal/model/Security.go
package model

import (
    "gorm.io/gorm"
)

//==============================================================
// Security Declaration
//==============================================================
type Security struct {
    gorm.Model
     Symbol                                    string
    Isin                                    string
    Cusip                                    string
    Currency                                    string
     Positions           []Position `gorm:"foreignKey:PositionsFromSecurityId"`
     Trades           []Trade `gorm:"foreignKey:TradesFromSecurityId"`
     Orders           []TradeOrder `gorm:"foreignKey:OrdersFromSecurityId"`
    SecurityType                      SecurityType

// parent associations as their child

}

File: evidence/generated-systems/rubyonrails/fintech-on-rails/fintechonrails/app/models/security.rb
class Security < ApplicationRecord
  enum SecurityType: [:Equity, :Bond, :ETF, :MutualFund, :Derivative, :Crypto]


  has_many :Positions, class_name: 'Position'
  has_many :Trades, class_name: 'Trade'
  has_many :Orders, class_name: 'TradeOrder'

end
